src/aglet/target.nim

```diff
@@ -34,19 +36,38 @@ proc height*(target: Target): int {.inline.} =
   ## Returns the height of the target.
   target.size.y
 
-proc clearColor*(target: Target, color: Rgba32f) {.inline.} =
+proc scissor(target: Target, scissor: Option[Recti] = Recti.none) =
+  if scissor.isSome():
+    let scissor = scissor.get
+    target.gl.scissor(scissor.x, scissor.y, scissor.width, scissor.height)
+  else:
+    target.gl.scissor(0, 0, target.size.x, target.size.y)
+
+proc clearColor*(target: Target, color: Rgba32f, scissor: Option[Recti] = Recti.none) {.inline.} =
```

Comment:
I'm not sure I like this solution with `scissor` being a parameter to `clear{Color,Depth,Stencil}`. If more bugs like this appear, it would make sense to hoist the extra parameters into an object of their own, kind of like `DrawParams`, but I think more often than not you want to share the scissor rectangle between subsequent calls to `clear{Color,Depth,Stencil}`. This is an idea for the future, but maybe we should replace `clearColor`, `clearDepth`, and `clearStencil` with a unified `clear` proc that accepts a single `ClearParams` object, which houses optional fields for the color, depth, and stencil values, as well as the optional scissor rectangle.